src/music/Rhythm.ts

```diff
@@ -0,0 +1,52 @@
+import { Random } from "../util/Random.js";
+import { NonEmptyArray } from "../util/Arrays.js";
+
+export default class Rhythm {
+
+    subdivision: number[];
+    _beats: number;
+
+    constructor(beats: number = 8) {
+        this.subdivision = [];
+        this._beats = beats;
+        this.generateNewSubdivision(beats);
+    }
+
+    get beats(): number {
+        return this._beats;
+    }
+
+    // combine subdivisions at random
+    generateTies(): number[] {
+        let i: number = 0;
+        const r: number[] = [];
+        while (i < this.subdivision.length) {
+            if (i < this.subdivision.length - 1 && Random.bool(0.5)) {
+                r.push(this.subdivision[i] + this.subdivision[i + 1]);
+                i += 2;
+            } else {
+                r.push(this.subdivision[i]);
+                i++;
+            }
+        }
+        return r;
+    }
+
+    generateNewSubdivision(beats: number): void {
+        this._beats = beats;
+        this.subdivision = [];
+        const divisions = [2, 3];
```

Comment:
Why is this just a `const` in a method? If there's a reason why you made it this, document it. If it doesn't need to be in this function, move it up.